digest.go

```diff
@@ -258,6 +270,7 @@ func (a *DigestAuth) NewContext(ctx context.Context, r *http.Request) context.Co
 			fmt.Sprintf(`Digest realm="%s", nonce="%s", opaque="%s", algorithm="MD5", qop="auth"`,
 				a.Realm, nonce, a.Opaque))
 	}
+	a.mutex.Unlock()
```

Comment:
Idiomatic go would be to put this in a `defer a.mutex.Unlock()` and move it directly. WDYT?

Reply:
You right, I thought that `context.WithValue (ctx, infoKey, info)` was taking time. And I didn't want to lock too long